src/main.js

```diff
@@ -209,10 +214,11 @@ export async function convertToQuorum(argv) {
   console.log('Conversion to Quorum Vat complete.');
 }
 
-export async function buildArgv(vat, argvJSON, readBaseFile, vatEndowments) {
+export function buildArgv(vat, argvJSON, readBaseFile, vatEndowments) {
   const argv = vat.makeEmptyObject(); // realm-side object
   const descs = JSON.parse(argvJSON);
-  for (const name of Object.getOwnPropertyNames(descs)) {
+  const names = Object.getOwnPropertyNames(descs);
+  names.forEach(async name => {
```

Comment:
what exactly will this do when the enclosing function is not `async`? I'm wondering if it will create a Promise for each pass (which doesn't fire until e.g. `readBaseFile` finishes, good) but then either discard the promise (bad) or have `names.forEach` do something clever like return a promise that doesn't fire until the entire loop fires, but then we're throwing away *that* promise (also bad).

This might require an experiment to see what the right syntax is. The overall requirement is that the caller of buildArgv gets to know exactly when it's safe to use the `argv` object, and that includes having the contents of files be fully available. (we don't have a mechanism for `argv` to contain promises that later code can wait for.. they're supposed to contain fully-formed values)

Reply:
overall maybe we can disable the warnings and leave this as a for..of loop ?

Reply:
I'm going to revert this for now.